src/sentry/eventstore/base.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import absolute_import
+
+from sentry import nodestore
+from sentry.utils.services import Service
+
+
+class EventStorage(Service):
+    __all__ = (
+        'minimal_columns',
+        'full_columns',
+        'get_event_by_id',
+        'get_events',
+        'get_prev_event_id',
+        'get_next_event_id',
+        'bind_nodes',
+    )
+
+    # The minimal list of columns we need to get from snuba to bootstrap an
+    # event. If the client is planning on loading the entire event body from
+    # nodestore anyway, we may as well only fetch the minimum from snuba to
+    # avoid duplicated work.
+    minimal_columns = [
+        'event_id',
+        'group_id',
+        'project_id',
+        'timestamp',
+    ]
+
+    # A list of all useful columns we can get from snuba.
+    full_columns = minimal_columns + [
+        'culprit',
+        'location',
+        'message',
+        'platform',
+        'title',
+        'type',
+
+        # Required to provide snuba-only tags
+        'tags.key',
+        'tags.value',
+
+        # Required to provide snuba-only 'user' interface
+        'email',
+        'ip_address',
+        'user_id',
+        'username',
+    ]
+
+    def get_event_by_id(self, project_id, event_id, cols):
```

Comment:
If we allow and encourage people to specify the columns they want to load, we should represent them with something safer than simple strings. Using an enum for example makes it easy to avoid typos that would be visible only at runtime. It will also make things easier the day  we will have type hints.

Reply:
Could we have column 'profiles'. I suspect we have a few commonly used column groups and putting them into enums would let us save repetition and discourage ❄️ usage.

Reply:
Yeah, I think we'd keep the "selected_columns" and "minimal_columns" lists relatively intact https://github.com/getsentry/sentry/blob/master/src/sentry/models/event.py#L433-L458 and use these wherever possible. I'll update this PR to include the default values for these fields.

I'm not sure if we can use enum since users can send us any random string as a custom tag value?

Reply:
Updated to use Enum